compiler/extccomp.nim

```diff
@@ -14,7 +14,7 @@
 
 import ropes, platform, condsyms, options, msgs, lineinfos, pathutils
 
-import os, strutils, osproc, std/sha1, streams, sequtils, times, strtabs, json
+import std/[os, strutils, osproc, sha1, streams, sequtils, times, strtabs, json, jsonutils, sugar]
```

Comment:
No. If you want to improve this, give us a "jsonbuilder" module first. One that doesn't repeat json's mistakes.

Reply:
@araq are you referring to https://github.com/timotheecour/Nim/issues/745 (serialize/deserialize directly from string instead of from intermediate JsonNode) or to something else? If so, yes I agree https://github.com/timotheecour/Nim/issues/745 is needed down the line for performance but it would be a 1 liner change on top of this PR once that feature lands. If you're worried about performance I can make a benchmark but I don't see a reason why this PR would create any meaningful overhead.

jsonbuilder sounds like it'd be instead similar to the `%*` operator (`%* {"age":35, "pi":3.1415}`), but that's worse than what I'm doing as there's no spec/type-safety and serialization/deserialization would need more syncrhonization.

this PR is a big code simplification